digital_reader.go

```diff
@@ -5,58 +5,59 @@ import (
 	"time"
 )
 
-const DefaultDebounceDelay = time.Duration(50 * time.Millisecond)
+const defaultDebounceDelay = time.Duration(50 * time.Millisecond)
+
+type callback func(machine.Pin)
 
 // DigitalReader is an interface for common digital inputs methods.
 type DigitalReader interface {
-	Handler(func(machine.Pin))
-	HandlerWithDebounce(func(machine.Pin), time.Duration)
+	Handler(callback)
```

Comment:
This is an odd API design choice. having an unexported type in an exported interface. Do you want to define semantics on the `callback` type? Also be careful with how you use this interface- it's present in digitalInput and button as an exported embedded field on an unexported type which are present as an exported field on `EuroPi`, which means the underlying `DigitalReader` on the `button` and `digitalInput` is accesible to users to reference. I'd consider hiding this in `button` and `digitalInput` behind an unexported field and allowing users to set them on initialization.